swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/Store.swift

```diff
@@ -301,13 +299,36 @@ public final class Store: ObservableObject {
     }
 #endif
 
+    self.configuration = configuration
+
     if Telemetry.firezoneId == nil {
       Telemetry.firezoneId = configuration?.firezoneId
     }
 
+    try await updateAppService()
+
     return configuration
   }
 
+  // Register / unregister our launch service based on configuration. This is a major pain to do on macOS 12 and below,
+  // so this feature only enabled for macOS 13 and higher given the tiny Firezone installbase for macOS 12.
+  private func updateAppService() async throws {
+#if os(macOS)
```

Comment:
I'd suggest refactoring this to early exits:

```
if !os(macOS)
    return

if !#available(macOS 13.0, *)
    return
```

Might not even need the first item then.

Reply:
Unfortunately the macros don't support that.

Reply:
We're stuck with if/else for these.

<img width="1196" alt="Screenshot 2025-05-16 at 2 56 13 PM" src="https://github.com/user-attachments/assets/d617c5b3-f4da-4657-8de0-834ed2eb9ce9" />
<img width="1197" alt="Screenshot 2025-05-16 at 2 56 01 PM" src="https://github.com/user-attachments/assets/8490f4a9-9b0f-43d0-9093-16ddf84a6ab3" />

Reply:
Oh, it is a macro `if`. Maybe we can gate the entire function and have an empty fallback instead?

Reply:
The function signature would have to be different, otherwise we'd get a warning.

Reply:
We could gate both the function and the caller I suppose. We may end up refactoring this anyway to have macOS and iOS variants down the road.

Reply:
> The function signature would have to be different, otherwise we'd get a compiler warning.

Which part of `updateAppService` needs to change? Seems to me like that could stay the same.

Happy to roll with what we have though, not super important :)

Reply:
Was able to update it to gate the entire function and just have an empty body for non-macOS

Reply:
You didn't seem to have pushed it though :)

It is already in the queue so further pushes are denied by GitHub, maybe you ran into that?

Reply:
This ended up getting removed in #9196